server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/redirect/ProcessRedirectNotify.java

```diff
@@ -0,0 +1,175 @@
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.transport.mailets.redirect;
+
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.core.MailImpl;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.StringUtils;
+
+public class ProcessRedirectNotify {
+
+    public static ProcessRedirectNotify from(RedirectNotify mailet) {
+        return new ProcessRedirectNotify(mailet);
+    }
+
+    private final RedirectNotify mailet;
+    private final boolean isDebug;
+
+    private ProcessRedirectNotify(RedirectNotify mailet) {
+        this.mailet = mailet;
+        this.isDebug = mailet.getInitParameters().isDebug();
+    }
+
+    public void process(Mail originalMail) throws MessagingException {
+
+        // duplicates the Mail object, to be able to modify the new mail keeping
+        // the original untouched
+        MailImpl newMail = new MailImpl(originalMail);
+        try {
+            MailModifier mailModifier = MailModifier.builder()
+                    .mailet(mailet)
+                    .mail(newMail)
+                    .dns(mailet.getDNSService())
+                    .build();
+            mailModifier.setRemoteAddr();
+            mailModifier.setRemoteHost();
+
+            if (isDebug) {
+                mailet.log("New mail - sender: " + newMail.getSender() + ", recipients: " + StringUtils.arrayToString(newMail.getRecipients().toArray()) + ", name: " + newMail.getName() + ", remoteHost: " + newMail.getRemoteHost() + ", remoteAddr: " + newMail.getRemoteAddr() + ", state: " + newMail.getState()
+                        + ", lastUpdated: " + newMail.getLastUpdated() + ", errorMessage: " + newMail.getErrorMessage());
+            }
+
+            // Create the message
+            boolean keepMessageId = false;
```

Comment:
Re-affectation spotted.

In a keepMessageId() method : 

```
if (mailet.getInitParameters().getInLineType().equals(TypeCode.UNALTERED)) {
    return true;
}
return false;
```

In here you can keep the createAltered / createUnaltered stuff.